packages/expo-av/ios/EXAV/Video/EXVideoView.m

```diff
@@ -241,35 +241,18 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
       }
     }
   } else if (object == _playerViewController && [keyPath isEqualToString:EXVideoBoundsKeyPath]) {
-    UIViewController *presentedViewController = [[_moduleRegistry getModuleImplementingProtocol:@protocol(UMUtilitiesInterface)] currentViewController];
-    if (presentedViewController == nil) {
-      return;
-    }
-
-    // For a short explanation on why we're detecting fullscreen changes in such an extraordinary way
-    // see https://stackoverflow.com/questions/36323259/detect-video-playing-full-screen-in-portrait-or-landscape/36388184#36388184
-
-    // We may be presenting a fullscreen player for this video item
-    UIViewController *fullscreenViewController;
-
-    if ([[presentedViewController.class description] isEqualToString:EXAVFullScreenViewControllerClassName]) {
-      // RCTPresentedViewController() is fullscreen
-       fullscreenViewController = presentedViewController;
-    } else if (presentedViewController.presentedViewController != nil && [[presentedViewController.presentedViewController.class description] isEqualToString:EXAVFullScreenViewControllerClassName]) {
-      // RCTPresentedViewController().presentedViewController is fullscreen
-      fullscreenViewController = presentedViewController.presentedViewController;
-    }
-
-    if (fullscreenViewController.isBeingDismissed && _fullscreenPlayerPresented && _nativeFullscreenPlayerViewController == fullscreenViewController) {
+    
+    CGRect viewBounds = [change[@"new"] CGRectValue];
+    CGRect screen = [[UIScreen mainScreen] bounds];
+    
+    if (viewBounds.size.height != screen.size.height && viewBounds.size.width != screen.size.width && _fullscreenPlayerPresented) {
       // Fullscreen player is being dismissed
       _fullscreenPlayerPresented = false;
-      _nativeFullscreenPlayerViewController = nil;
       [self _callFullscreenCallbackForUpdate:EXVideoFullscreenUpdatePlayerWillDismiss];
       [self _callFullscreenCallbackForUpdate:EXVideoFullscreenUpdatePlayerDidDismiss];
-    } else if (fullscreenViewController.isBeingPresented && !_fullscreenPlayerPresented && _nativeFullscreenPlayerViewController == nil) {
+    } else if (viewBounds.size.height == screen.size.height && viewBounds.size.width == screen.size.width && !_fullscreenPlayerPresented) {
       // Fullscreen player is being presented
       _fullscreenPlayerPresented = true;
-      _nativeFullscreenPlayerViewController = fullscreenViewController;
```

Comment:
I guess we could remove this variable altogether now.

Reply:
Yes we can!